tests/tagService_test.go

```diff
@@ -0,0 +1,355 @@
+			testServer := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				w.WriteHeader(testCase.statusCode)
+				_, writeError := w.Write([]byte(testCase.data))
+				if writeError != nil {
+					fmt.Printf("Error: %s", writeError)
+				}
+			}))
+			defer testServer.Close()
+			client := gointelowl.MakeNewIntelOwlClient(
+				&gointelowl.IntelOwlClientOptions{
+					Url:         testServer.URL,
+					Token:       "test-token",
+					Certificate: "",
+				},
+				nil,
+			)
+			ctx := context.Background()
+			gottenTag, err := client.TagService.Get(ctx, testCase.input)
+			if testCase.statusCode < http.StatusOK || testCase.statusCode >= http.StatusBadRequest {
+				diff := cmp.Diff(testCase.want, err)
+				if diff != "" {
+					t.Fatalf(diff)
+				}
+			} else {
+				diff := cmp.Diff(testCase.want, gottenTag)
+				if diff != "" {
+					t.Fatalf(diff)
+				}
+			}
+		})
+	}
+}
+
+func TestTagServiceCreate(t *testing.T) {
+	// *table test case
+	testCases := map[string]struct {
+		input      *gointelowl.TagParams
+		data       string
+		statusCode int
+		want       interface{}
+	}{
+		"simple": {
+			input: &gointelowl.TagParams{
+				Label: "TEST TAG",
+				Color: "#fffff",
+			},
+			data:       `{"id": 1,"label": "TEST TAG","color": "#fffff"}`,
+			statusCode: 200,
+			want: &gointelowl.Tag{
+				ID:    1,
+				Label: "TEST TAG",
+				Color: "#fffff",
+			},
+		},
+		"duplicate": {
+			input: &gointelowl.TagParams{
+				Label: "TEST TAG",
+				Color: "#fffff",
+			},
+			data:       `{"label":["tag with this label already exists."]}`,
+			statusCode: 400,
+			want: &gointelowl.IntelOwlError{
+				StatusCode: 400,
+				Data:       []byte(`{"label":["tag with this label already exists."]}`),
+			},
+		},
+	}
+	for name, testCase := range testCases {
+		//* Subtest
+		t.Run(name, func(t *testing.T) {
+			testServer := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				w.WriteHeader(testCase.statusCode)
+				_, err := w.Write([]byte(testCase.data))
+				if err != nil {
+					fmt.Printf("Error: %s", err)
+				}
+
+			}))
+			defer testServer.Close()
+			client := gointelowl.MakeNewIntelOwlClient(
+				&gointelowl.IntelOwlClientOptions{
+					Url:         testServer.URL,
+					Token:       "test-token",
+					Certificate: "",
+				},
+				nil,
+			)
+			ctx := context.Background()
+			gottenTag, err := client.TagService.Create(ctx, testCase.input)
+			if testCase.statusCode < http.StatusOK || testCase.statusCode >= http.StatusBadRequest {
+				diff := cmp.Diff(testCase.want, err)
+				if diff != "" {
+					t.Fatalf(diff)
+				}
+			} else {
+				diff := cmp.Diff(testCase.want, gottenTag)
+				if diff != "" {
+					t.Fatalf(diff)
+				}
+			}
+		})
+	}
+}
+
+func TestTagServiceUpdate(t *testing.T) {
+	// *table test case
+	testCases := map[string]struct {
+		input      gointelowl.Tag
+		data       string
+		statusCode int
+		want       interface{}
+	}{
```

Comment:
Can't we extract this map into a type/struct on the file level instead of having to define individually inside each method?

Reply:
Sure! That would be better and every test of each service object will be like this?

Reply:
@eshaan7 and @sp35 so casting an interface is bad idea reference [here](https://stackoverflow.com/questions/50939497/golang-cast-interface-to-struct) . I was thinking of making one `Test` struct so that we could use it in every service object test. Any suggestions?